src/main/java/seedu/tasklist/model/task/EndTime.java

```diff
@@ -37,43 +41,61 @@ else if(dates.get(0).getDates().isEmpty()){
     			throw new IllegalValueException("End time is invalid!");
     		}
     		else{
-    			this.endTime = dates.get(0).getDates().get(0);
+    			endtime.setTime(dates.get(0).getDates().get(0));
     		}
     	}
     	else{
-    		this.endTime = new Date(0);
+    	      endtime.setTime(new Date(0));
     	}
     }
 
     @Override
     public String toString() {
-    	if(endTime.equals(new Date(0))){
+    	if(endtime.getTime().equals(new Date(0))){
     		return (new Date(0)).toString();
     	}
     	else{
-    		return endTime.toString();
+    		return endtime.getTime().toString();
     	}
     }
     
     public String toCardString() {
-    	if(endTime.equals(new Date(0))){
+    	if(endtime.equals(new Date(0))){
     		return "-";
     	}
     	else{
-    		return endTime.toString();
+    		DateFormat df = new SimpleDateFormat("dd/MM/yyyy HH:mm");
```

Comment:
Unnecessary manipulation, and also truncates the minutes. SimpleDateFormat allows you to display hours in 12-h format and add the a.m./p.m. marker, so all you need is actually the first 3 lines of this block